### ISSUE TITLE
Client.Timeout exceeded while awaiting headers

### ISSUE THREAD

**Issue by AleksandrOrlov** [docker/for-win#1534]
### Expected behavior

Download Image and create container

### Actual behavior

Unable to find image 'hello-world:latest' locally
docker: Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
See 'docker run --help'.

### Information
Diagnostic Id: AD27B128-4EC0-4888-9470-22E42CD7DF37/2018-01-12_13-49-18

I try all variations of solution, and they not working for me:
1) Set DNS to Fixed 8.8.8.8
2) Reset factory/Reinstall docker/Reboot docker or computer

### Steps to reproduce the behavior

  1. Execute command in console: docker run hello-world

``

**Comment by vissihaus** [comment]
Same thing here...

**Comment by thomao** [comment]
I also experience the same issue...

**Comment by helysousa** [comment]
I had the same problem and following @jasonbivins' instructions in the post above I noticed that I couldn't also run `docker login`. After fixing the username and password issue I was ready to run `docker login` and `docker run hello-world`.

**Comment by agui2200** [comment]
Same thing here... too
but i try switch docker containers can fix him,
but can't pull linux platform images...WTF?
and i try "docker login" is same too.
and i try switch docker config ,some time no change (i restart docker from windows).
platform : windows 1709
docker :  17.12.0-ce

**Comment by pavelpp** [comment]
interesting enough same docker for win version is working on my work computer, but fails on my home desktop. Both are running Windows 10. login and docker helloworld are failing. Restarts/resets are not helping. Tried disabling system firewall - no difference.

**Comment by haruza** [comment]
Same thing here... too
Tried docker login and restart windows, hyper-v and reinstalled docker

**Comment by dElogics** [comment]
Looks like the registry or Internet is slow and there's no way to set this timeout. That explains the random behavior.

**Comment by pavelpp** [comment]
@dElogics it's not random. Two machines connected to the same router, fast internet. One works another does not in 100% cases.

**Comment by marcusholl** [comment]
Same thing on our side.

**Comment by IngoVals** [comment]
I'm having the same problem and for me it doesn't work with Linux containers but works fine with Windows containers.

**Comment by betterdelta** [comment]
Seems to be resolved for me when disabling experimental features.

**Comment by IngoVals** [comment]
Yeah this is a known bug, [more info here.](https://github.com/docker/for-win/issues/611#issuecomment-365023640)

**Comment by f-lopes** [comment]
Worked for me using @helysousa solution: `docker login [registry]`. Thanks.

**Comment by montao** [comment]
Doesn't work. This makes Docker completely useless for us.

**Comment by YaMolekula** [comment]
Same problem. I can't download ant image.

**Comment by garciasa** [comment]
Same issue. I've tried everything I've seen on Internet but nothing works :(

**Comment by dev-e** [comment]
I also confirm issue but only with docker.io repository. I am able to pull images correctly from other repos like quay,io, gcr.io, elastic.co. Not sure how to reproduce the issue, sometimes images are pulled, sometimes not on different hosts. Docker is behind corporate ssl proxy.
```
~ $ docker version
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.4
 Git commit:   9ee9f40
 Built:        Thu Apr 26 04:27:49 2018
 OS/Arch:      linux/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.4
  Git commit:   9ee9f40
  Built:        Thu Apr 26 04:27:49 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

**Comment by cakirilker** [comment]
Docker was working fine and all the sudden I got same error when pulling images. 
````
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
````
Changing DNS to 8.8.8.8 / 8.8.4.4 and restarting docker solved my issue.

**Comment by sumitdubey022229** [comment]
I also faced same problem.
I tried many solutions but what solved my problem was this one-
If you are behind a proxy then follow these steps:

1. Remove network proxy in Network &Settings.
2.Configure Docker Proxy in following format 
      http://username:password@proxy-server:proxy-port  for http proxy 
replace http by https for https proxy.

eg: http://sumitDubey:password@proxy@company.com:8080

**Comment by monsdar** [comment]
Same issue over here. No proxy. DNS is set to 8.8.8.8 and 8.8.4.4 for all adapters I could find (Wireless, LAN, vEthernet, ...).

I was able to do `docker login` just yesterday. Today it's not possible anymore. Same network, same computer, same installation, same docker version. Gonna update this post when I find something that's working.

**Comment by jasclarke24** [comment]
I get same exception behind a proxy. I can **docker login** with no wildcards in my bypass list and an entry in bypass list for proper domain. Put a pull from internal repo inside of proxy fails with _(Client.Timeout exceeded while awaiting headers)_

In addition, tried doing this with _Experimental features_ disabled which had no affect on failure. 

Also tried with "_No Proxy_" set and still no effect on failure. I can login but not pull.

Switched DNS to _Fixed: 8.8.8.8_ and could no longer log in. Can login with above settings when DNS  _Automatic_ selected.

**Comment by jasclarke24** [comment]
I got it working with the following settings

1) Proper Manual Proxy Configuration with proper domain in bypass list (no wildcards) _mydomain.com_
2) Experimental Features disabled (this may or may not be an effect)
3) DNS Server set to Automatic

Note: It may have been an issue mostly for me regarding the path to my actual image in the repo

**Comment by jefjos** [comment]
I faced this issue in Windows and turning off and turning back on Windows feature Hyper-V helped me.

**Comment by matheuslemke** [comment]
Missing proxy configuration?

`mkdir -p /etc/systemd/system/docker.service.d`
`nano /etc/systemd/system/docker.service.d/http-proxy.conf`

[Service]
Environment="HTTP_PROXY=http://USER:PASSWD@SERVER:PORT/"
Environment="HTTPS_PROXY=http://USER:PASSWD@SERVER:PORT/"

`systemctl daemon-reload`
`systemctl restart docker`

**Comment by spawn1978** [comment]
Thanks!  It worked

**Comment by MPJHorner** [comment]
I was getting the same errors (on Windows) 

```
Version 18.03.1-ce-win65 (17513)
Channel: stable
93354b3
```

Restarted docker, works fine.

**Comment by PranavKD** [comment]
The following solution worked for me: 
1. create new virtual switch manager and set it type as external.
2. Click on MobyLinuxVM settings and change its network adapter to the newly created virtual switch manager.
2. Open your docker settings and go to network tab. Select Fixed option under DNS Server.(The value will be 8.8.8.8)

**Comment by matheuslemke** [comment]
@AleksandrOrlov can you close this issue please? 

Thanks

**Comment by jainishshah17** [comment]
Same here. I am using docker on ubuntu.

**Comment by tomtom30** [comment]
For me too, with ubuntu server.

**Comment by Femina** [comment]
I got the same error. But, after Docker Login everything seems to be working fine.

**Comment by jokerdrake** [comment]
@cakirilker Thx, I solved this problem.

**Comment by d3vit** [comment]
Logging in and restarting Docker fixed this issue for me

**Comment by byondrnd** [comment]
I am experiencing the same problem. It used to work perfectly until recently and stopped. Blocking me from doing anything with docker for windows.
As far as version data goes:
**.\docker.exe version**
Client: Docker Engine - Community
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        4d60db4
 Built:             Wed Nov  7 00:47:51 2018
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:55:00 2018
  OS/Arch:          linux/amd64
  Experimental:     false


I tried all of the suggested solutions and workarounds suggested in this issue...
Including, docker reinstallation, dns configurations, reboots, switching networks (LTE, Wifi, Ether, etc...), different ISPs, etc...
None worked.

It is obviously severely affecting numerous installations.

**Comment by cedmcgee** [comment]
> I had the same problem and following @jasonbivins' instructions in the post above I noticed that I couldn't also run `docker login`. After fixing the username and password issue I was ready to run `docker login` and `docker run hello-world`. Thanks, @jasonbivins for the insight provided.

What was the issue ?

**Comment by matheuslemke** [comment]
> > I had the same problem and following @jasonbivins' instructions in the post above I noticed that I couldn't also run `docker login`. After fixing the username and password issue I was ready to run `docker login` and `docker run hello-world`. Thanks, @jasonbivins for the insight provided.
> 
> What was the issue ?

Proxy configuration

**Comment by WattsC-90** [comment]
> Missing proxy configuration?
> 
> `mkdir -p /etc/systemd/system/docker.service.d`
> `nano /etc/systemd/system/docker.service.d/http-proxy.conf`
> 
> [Service]
> Environment="HTTP_PROXY=http://USER:PASSWD@SERVER:PORT/"
> Environment="HTTPS_PROXY=http://USER:PASSWD@SERVER:PORT/"
> 
> `systemctl daemon-reload`
> `systemctl restart docker`

While this solves the issue, you always need to reload the daemon and restart the docker service, how can I get it to work on boot? if the service is enabled, it still doesnt load these proxy settings until it is `reload`ed and `restarted`

**Comment by jeffdockergit** [comment]
I have no proxy and tried to set the environment for docker.service to.
[Service]
Environment="NO_PROXY=localhost,127.0.0.1"

systemctl daemon-reload
systemctl reload docker
systemctl show --property Environment docker
Environment=NO_PROXY=localhost,127.0.0.1

cat /etc/resolv.conf
# Generated by NetworkManager
search lab.local
nameserver 1.1.1.1.               
nameserver 203.144.193.117      # ISP dns

Still no luck.

docker --debug run hello-world
Unable to find image 'hello-world:latest' locally
docker: Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
See 'docker run --help'.

docker system info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 18.09.3
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: e6b3f5632f50dbc4e9cb6288d911bf4f5e95b18e
runc version: 6635b4f0c6af3810594d2770f662f34ddc15b40d
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-957.5.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.795GiB
Name: chefdocker.lab.local
ID: SRTC:IXVL:PRJY:N4DI:MKYM:7XJI:GS7Y:JOPN:XN4R:HGZ4:QQDA:Q3XG
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
No Proxy: localhost,127.0.0.1
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine

Any clue?

**Comment by jigarakatidus** [comment]
I use docker-compose for one of my project which creates the docker `network` for that project. I did the following to make things work. 
- `cd /my/project`
- `docker-compose down` (removes network)
- `docker-machine stop`
- `docker-machine start`

Sorry, I don't know what happens behind the scene. But this worked for me.

May be pruning network can help
- `docker network prune`

**Comment by shawn-ogg** [comment]
I have the same issue. After a fresh boot I get those timeout errors (for any network related command). Only after a restart of docker-desktop everything works fine. My guess would be that all those suggestions of changing one google dns with another are not so important - The restarting of the docker service was it!

Anyway this is a pretty annoying behavior. I had the same behavior on the edge and the sable channel.

**Comment by WowMuchName** [comment]
**Not** behind a proxy. Changing DNS to 8.8.8.8, then changing it back to automatic made the error go away.

**Comment by HendricksK** [comment]
had the same issue, my DNS was set to 8.8.8.8 snd I set it to dynamic and that seems to have fixed the issue, so similar fix to @WowMuchName

**Comment by jtanaka** [comment]
The solution by @WowMuchName works for me too.
I knew how to change DNS with the image below.
![image](https://cdn-ak.f.st-hatena.com/images/fotolife/s/satoysan/20170401/20170401174535.png)
Image from https://satoyashiki.hatenablog.com/entry/2017/04/01/175604

**Comment by Genovo** [comment]
or...it could just be timing out because dockerhub server is busy downloading 100MB images all day....

**Comment by cpurtlebaugh** [comment]
Simply restarting docker worked for me but note there's no proxy involved

**Comment by aliwo** [comment]
> 
> 
> Docker was working fine and all the sudden I got same error when pulling images.
> 
> ```
> Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
> ```
> 
> Changing DNS to 8.8.8.8 / 8.8.4.4 and restarting docker solved my issue.

Thanks! It works like charm! Btw, Why is this working?

**Comment by gcartier94** [comment]
The fix for me was to select "Switch to Windows Containers".

**Comment by KrazyMax** [comment]
> Missing proxy configuration?
> 
> `mkdir -p /etc/systemd/system/docker.service.d`
> `nano /etc/systemd/system/docker.service.d/http-proxy.conf`
> 
> [Service]
> Environment="HTTP_PROXY=http://USER:PASSWD@SERVER:PORT/"
> Environment="HTTPS_PROXY=http://USER:PASSWD@SERVER:PORT/"
> 
> `systemctl daemon-reload`
> `systemctl restart docker`

Thanks for this answer! It helped me and solved my proxy issue. I don't understand the [Docker documentation](https://docs.docker.com/network/proxy/#configure-the-docker-client) I followed then... I did what was specified, but does not work (they ask user to create another file in another directory than those you wrote here). Docker documentation is a bit messy in my opinion!

**Comment by sbabaei** [comment]
I had this problem, and it is because my country in under US sanctions with sorry, I use VPN and it is solved for me.

**Comment by bhlevca** [comment]
> Missing proxy configuration?
> 
> `mkdir -p /etc/systemd/system/docker.service.d`
> `nano /etc/systemd/system/docker.service.d/http-proxy.conf`
> 
> [Service]
> Environment="HTTP_PROXY=http://USER:PASSWD@SERVER:PORT/"
> Environment="HTTPS_PROXY=http://USER:PASSWD@SERVER:PORT/"
> 
> `systemctl daemon-reload`
> `systemctl restart docker`

Thanks, it pointed me to the right path. However on OpenSuSE I needed to edit the 
`/usr/lib/systemd/system/docker.service` file to include the the proxy env variable, here is the entire file content:

```
[Unit]
Description=Docker Application Container Engine
Documentation=http://docs.docker.com
After=network.target lvm2-monitor.service SuSEfirewall2.service

[Service]
EnvironmentFile=/etc/sysconfig/docker

# While Docker has support for socket activation (-H fd://), this is not
# enabled by default because enabling socket activation means that on boot your
# containers won't start until someone tries to administer the Docker daemon.
Type=notify
ExecStart=/usr/bin/dockerd --add-runtime oci=/usr/sbin/docker-runc $DOCKER_NETWORK_OPTIONS $DOCKER_OPTS
ExecReload=/bin/kill -s HUP $MAINPID

# Having non-zero Limit*s causes performance problems due to accounting overhead
# in the kernel. We recommend using cgroups to do container-local accounting.
LimitNOFILE=1048576
LimitNPROC=infinity
LimitCORE=infinity

# Uncomment TasksMax if your systemd version supports it.
# Only systemd 226 and above support this property.
TasksMax=infinity

# Set delegate yes so that systemd does not reset the cgroups of docker containers
# Only systemd 218 and above support this property.
Delegate=yes

# Kill only the docker process, not all processes in the cgroup.
KillMode=process

# Restart the docker process if it exits prematurely.
Restart=on-failure
StartLimitBurst=3
StartLimitInterval=60s

[Install]
WantedBy=multi-user.target


[Service]
Environment="HTTP_PROXY=http://hlevcabo:Raluca1011204.40.194.129:3128/"
Environment="HTTPS_PROXY=http://hlevcabo:Raluca1011204.40.194.129:3128/"


```

**Comment by tartansandal** [comment]
As suggested by @jigarakatidus above:

`docker network prune`

works for me.  Perhaps networks just stack up and cause the timeout?

**Comment by hsmiranda** [comment]
i'm resolved this problem with setting proxy variables correct, oc or kubectl utilize proxy variables for connect to api even on localhost.

**Comment by kg-2** [comment]
disabling my VPN solved it for me

**Comment by ra-coder** [comment]
for me helped 

`sudo systemctl restart docker
`

on host with private registry

**Comment by will7455** [comment]
> ### Expected behavior
> Download Image and create container
> 
> ### Actual behavior
> Unable to find image 'hello-world:latest' locally
> docker: Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
> See 'docker run --help'.
> 
> ### Information
> Diagnostic Id: AD27B128-4EC0-4888-9470-22E42CD7DF37/2018-01-12_13-49-18
> 
> I try all variations of solution, and they not working for me:
> 
> 1. Set DNS to Fixed 8.8.8.8
> 2. Reset factory/Reinstall docker/Reboot docker or computer
> 
> ### Steps to reproduce the behavior
> 1. Execute command in console: docker run hello-world
> 
> ``
HELLO! the problem most people is facing that the home computer is not setup like a SERVER! and guys that work around servers all day are giving advice for specific scenarios that dont apply to average "docker install windows / ubuntu doesnt work uninstall people" lol

Here is your fix
1 Install the resolvconf package.

`$ sudo apt-get update`

$ sudo apt install resolvconf

2 Edit /etc/resolvconf/resolv.conf.d/head and add the following:

**$ sudo nano /etc/resolvconf/resolv.conf.d/head**

# Make edits to /etc/resolvconf/resolv.conf.d/head.

nameserver 8.8.4.4
nameserver 8.8.8.8

**_PRESS CTRL X THEN PRESS YES TO SAVE_**
Restart the resolvconf service.

$ sudo service resolvconf restart

Fix should be permanent.

enjoy! 

-willy

**Comment by sharda1955** [comment]
> Hi @AleksandrOrlov
> Are you using a proxy? If so, what do you have setup in the GUI under settings? Can you run the ` docker login` command
> It also looks like the D: drive was unmounted or unreachable according to the logs
> 
> [13:11:11.235][ApiProxy ][Info ] time="2018-01-12T13:11:11+07:00" msg="proxy << POST /v1.35/images/create?fromImage=hello-world&tag=latest\n"
> [13:11:11.236][VpnKit ][Warning] vpnkit.exe: 9f31 Query:0 na:c:r:rn 0 <qs:registry-1.docker.io.localdomain. <AAAA|IN>> an: au: ad: lookup failed: no servers

hi ,

I am using proxy and not able to push the images from docker client to private repository, what could be the issue.

**Comment by sharda1955** [comment]
I ran docker login but getting error response from daemon " failed with status : 404 not found

**Comment by di118** [comment]
restarting my laptop did it for me

**Comment by sajibcu** [comment]
sudo systemctl start docker . this worked for me.

**Comment by elcolie** [comment]
I experience this in OSX. Changing the DNS to 8.8.8.8 solve. I suspect DNS is changed because of VPN

**Comment by marevalo10** [comment]
I have a Windows and I can figured this out by configuring the DNS as mentioned at the begining. The main issue was to know how to configure it. At the end, I have VirtualBox and docker uses one VM called default. I reviewed the configuration and saw it has 2 network interfaces but it didn't let me do any change. After this, I reviewed the configuration for some of the network interfaces (settings, network, change adapter option and then I selected the nw interface with the same name that VirtualBox default machine mentioned. 
Select properties, Internet Protocol v4, properties and then here you have DNS configuration empty!!:
![image](https://user-images.githubusercontent.com/19642437/79037458-94735d00-7c14-11ea-8c54-0ed5c95cd391.png)
